src/java.base/unix/native/libjava/ProcessImpl_md.c

```diff
@@ -562,15 +562,17 @@ spawnChild(JNIEnv *env, jobject process, ChildStuff *c, const char *helperpath)
 
     /* write the two structs and the data buffer */
     if (writeFully(c->childenv[1], (char *)&magic, sizeof(magic)) != sizeof(magic)) { // magic number first
-        return -1;
+        free(buf);
+	return -1;
```

Comment:
See jcheck warnings: there is `TAB` on this line, which should be replaced by 4-space indents.